servicenow/servicenow-mid-server.lua

```diff
@@ -0,0 +1,459 @@
+end
+
+--------------------------------------------------------------------------------
+-- EventQueue:add method
+-- @param e An event
+--------------------------------------------------------------------------------
+
+function EventQueue:add(e)
+
+  local type = "HOST"
+  local hostname = "Meta"
+  if e.host_id then
+    hostname = get_hostname(e.host_id)
+    if hostname == e.host_id then
+      if self.skip_anon_events ~= 1 then
+        broker_log:error(0, "EventQueue:add: unable to get hostname for host_id '" .. e.host_id .."'")
+        return false
+      else
+        broker_log:info(1, "EventQueue:add: ignoring that we can't resolve host_id '" .. e.host_id .."'. The event will be sent with the id only")
+      end
+    end
+  end
+
+  local service_description = "HOST"
+  if e.service_id then
+    type = "service"
+    service_description = get_service_description(e.host_id, e.service_id)
+    if service_description == e.service_id then
+      if self.skip_anon_events ~= 1 then
+        broker_log:error(0, "EventQueue:add: unable to get service_description for host_id '" .. e.host_id .."' and service_id '" .. e.service_id .."'")
+      else
+        broker_log:info(1, "EventQueue:add: ignoring that we can't resolve host_id '" .. e.host_id .."' and service_id '" .. e.service_id .."'")
+      end
+    end
+  elseif hostname == "Meta" then
+    service_description = e.output
+  end
+
+  -- Getting the host extended information
+  -- commented because we might not use them
+--  local xml_url = ''
+--  local xml_notes = ''
+--  local xml_service_severity = ''
+--  local xml_host_severity = ''
+--  if e.host_id then
+--    xml_host_severity = "<host_severity>" .. ifnil(broker_cache:get_severity(e.host_id), '0') .. "</host_severity>"
+--    if e.service_id then
+--      xml_url = ifnil(broker_cache:get_notes_url(e.host_id, e.service_id), 'no notes url for this service') 
+--      xml_service_severity = "<service_severity>" ..ifnil(broker_cache:get_severity(e.host_id, e.service_id), '0') .. "</service_severity>"
+--    else 
+--      xml_url = ifnil(broker_cache:get_action_url(e.host_id), 'no action url for this host') 
+--      xml_notes = "<host_notes>" .. ifnil(broker_cache:get_notes(e.host_id), 'OS not set') .. "</host_notes>"
+--    end
+--  end
+--
+--  local hgroups = broker_cache:get_hostgroups(d.host_id)
+--  if hgroups and #hgroups > 0 then
+--    grps = ""
+--    for idx = 1, #hgroups do
+--      grps = grps .. hgroups[idx].group_name .. ' '
+--    end
+--    labels = labels .. "host_groups=" .. grps .. ','
+--  end
+
+  -- Event to send
+  local event_to_send = {}
+  event_to_send["source"] = ifnil_or_empty(self.source, 'Centreon')
+  event_to_send["source_instance"] = ifnil_or_empty(self.source_instance, 'Centreon central server')
+  event_to_send["node"] = hostname
+  event_to_send["resource"] = service_description
+  event_to_send["metric_name"] = service_description
+  event_to_send["type"] = service_description
```

Comment:
I think it is better to use the following syntax:
event_to_send.source = ...
event_to_send.source_instance = ...